### PR TITLE
Assume filesystem version 1.0.0 if version link is missing

### LINKED ISSUE #323: links[Branch.Version] is undefined
# Summary

## Problem

Webnative throws an uncaught exception `TypeError: links[_pathJs.Branch.Version] is undefined`.

### Impact

This exception occurs with Vite, Parcel 2, and Webpack 5. Apps that use these bundlers/build tools will experience the exception.

## Solution

Not known. The exception appears to be thrown here: https://github.com/fission-suite/webnative/blob/8d405832892cb4cb8999269284ac55896cba9070/src/filesystem.ts#L100

# Detail

**Describe the bug**

Webnative throws an uncaught exception `TypeError: links[_pathJs.Branch.Version] is undefined`.

**To Reproduce**

 1.  Clone https://github.com/fission-suite/react-todomvc
 2. Upgrade webnative to `0.29.1` on the `vite`, `parcel`, and `webpack` branches
 3. `npm start` for local dev and `npm run build` for a build in each

**Expected behavior**

No errors in the console. The app works and can do TodoMVC things.

**Screenshots**

The `parcel` error:

branch-version-error

The errors from the other bundlers are similar, but more difficult to read.

### PR DESCRIPTION
**Pull request by matheus23**
Fixes #323 

Can you try this one out @bgins? :pray:

### CODE REVIEW

File: src/filesystem.ts [hunk 1]
```diff
@@ -97,12 +97,13 @@ export async function loadFileSystem(
 
 export async function checkVersion(filesystemCID: CID): Promise<void> {
   const links = await protocol.basic.getLinks(filesystemCID)
-  const versionStr = new TextDecoder().decode(await protocol.basic.getFile(links[Branch.Version].cid))
+  // if there's no version link, we assume it's from a 1.0.0-comatible version (from before ~ November 2020)
```
Comment (bgins) [inline src/filesystem.ts:100]: Is my filesystem that old?!? 😁

**Review by bgins — APPROVED** [review]
Yep! That worked.  Checked with all the bundlers mentioned in #323.

One thing I noticed -- a `version` wasn't added to https://brian.files.fission.name/ after I tested this out.

**Comment by matheus23** [comment]
> One thing I noticed -- a version wasn't added to https://brian.files.fission.name/ after I tested this out.

Even after a change and a DNSLink update? It *should* do that now.

File: src/filesystem.ts (hunk 1 above)
Comment (matheus23) [reply src/filesystem.ts:100]: Yep :)

```js
$ node --experimental-repl-await --no-warnings
Welcome to Node.js v16.11.1.
Type ".help" for more information.
> const fetch = (await import("node-fetch")).default
> const cborg = require("cborg")
> const response = await fetch("https://brian.files.fission.name/public/metadata")
> const arrayBuffer = await response.arrayBuffer()
> const metadata = cborg.decode(new Uint8Array(arrayBuffer))
> new Date(metadata.unixMeta.ctime).toDateString()
'Sun Oct 11 2020'
```

**Comment by bgins** [comment]
> > One thing I noticed -- a version wasn't added to https://brian.files.fission.name/ after I tested this out.
> 
> Even after a change and a DNSLink update? It _should_ do that now.

After making a change to the filesystem, the version was added. Thanks. 🙏

**Pull request merged**